command/debug/debug.go

```diff
@@ -320,212 +323,217 @@ func (c *cmd) captureStatic() error {
 	for output, v := range outputs {
 		marshaled, err := json.MarshalIndent(v, "", "\t")
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 
 		err = ioutil.WriteFile(fmt.Sprintf("%s/%s.json", c.output, output), marshaled, 0644)
 		if err != nil {
-			errors = multierror.Append(errors, err)
+			errs = multierror.Append(errs, err)
 		}
 	}
 
-	return errors
+	return errs
 }
 
-// captureDynamic blocks for the duration of the command
+// captureInterval blocks for the duration of the command
 // specified by the duration flag, capturing the dynamic
 // targets at the interval specified
-func (c *cmd) captureDynamic() error {
-	successChan := make(chan int64)
-	errCh := make(chan error)
+func (c *cmd) captureInterval() error {
+	intervalChn := time.NewTicker(c.interval)
+	defer intervalChn.Stop()
 	durationChn := time.After(c.duration)
 	intervalCount := 0
 
 	c.UI.Output(fmt.Sprintf("Beginning capture interval %s (%d)", time.Now().Local().String(), intervalCount))
 
-	// We'll wait for all of the targets configured to be
-	// captured before continuing
-	var wg sync.WaitGroup
+	err := captureShortLived(c)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case t := <-intervalChn.C:
+			intervalCount++
+			c.UI.Output(fmt.Sprintf("Capture successful %s (%d)", t.Local().String(), intervalCount))
```

Comment:
Maybe I'm missing something: why do we only output a success message after waiting? What if the `captureShortLived` call below fails?

Should this line maybe come after the call below succeeds?

Reply:
This will never be printed of `captureShortLived` return an error. The logic is a bit inverted because we want to run `captureShortLived ` before waiting for an `interval` long, this is why `captureShortLived` is called before the loop start first.

Reply:
Hmmm. Wouldn't it be clearer to the operator if they saw this right after the capture completes though?

I think I'd be confused if I was looking at debug logs and saw all the requests come in but nothing output for 30 seconds and then a message saying "success" right before the next interval and set of requests came in.

Perhaps this is how it worked before though? I don't think it's a huge deal it just puzzled me why we'd put it this way around rather than:

 1. initial `captureShortLived`
 2. Output success
 3. wait
 4. goto 1

So that the success message is shown right after the operation rather than at the end of the interval?

Reply:
I got your point, should be fixed now.